packages/api/src/resolvers/type/media-types.ts

```diff
@@ -1,12 +1,32 @@
-import { File as FileType } from '@unchainedshop/core-files';
+import { filesSettings, File as FileType, getFileAdapter } from '@unchainedshop/core-files';
 import { Context } from '../../context.js';
+import { checkAction } from '../../acl.js';
+import { actions } from '../../roles/index.js';
 
 export interface MediaHelperTypes {
-  url: (language: FileType, params: Record<string, any>, context: Context) => string;
+  url: (language: FileType, params: Record<string, any>, context: Context) => Promise<string>;
 }
 
 export const Media: MediaHelperTypes = {
-  url(root, params, { modules }) {
-    return modules.files.getUrl(root, params);
+  url: async (file, params, context) => {
+    const { modules } = context;
+    try {
+      await checkAction(context, actions.downloadFile, [file, params]);
+      const fileUploadAdapter = getFileAdapter();
+      const mediaUrl = modules.files.getUrl(file, params);
```

Comment:
getUrl should determine the URL and the File Adapter should be responsible to generate URL's that are accessible, else it won't work across different providers.  So that logic here for adding the signing query param should be moved to the adapter.

Reply:
still not resolved @Mikearaya